src/client/dfuse/ops/mknod.c

```diff
@@ -25,42 +25,48 @@
 #include "dfuse.h"
 
 void
-dfuse_cb_mkdir(fuse_req_t req, struct dfuse_inode_entry *parent,
+dfuse_cb_mknod(fuse_req_t req, struct dfuse_inode_entry *parent,
 	       const char *name, mode_t mode)
 {
 	struct dfuse_projection_info	*fs_handle = fuse_req_userdata(req);
-	struct dfuse_inode_entry	*ie = NULL;
-	int				rc;
+	struct dfuse_inode_entry	*ie;
+	int rc;
 
-	DFUSE_TRA_INFO(fs_handle,
-		       "Parent:%lu '%s'", parent->ie_stat.st_ino, name);
+	DFUSE_TRA_INFO(fs_handle, "Parent:%lu '%s'", parent->ie_stat.st_ino,
+		       name);
 
 	D_ALLOC_PTR(ie);
 	if (!ie)
 		D_GOTO(err, rc = ENOMEM);
 
 	DFUSE_TRA_UP(ie, parent, "inode");
 
-	DFUSE_TRA_DEBUG(ie, "directory '%s' mode 0%o", name, mode);
+	DFUSE_TRA_DEBUG(ie, "file '%s' mode 0%o", name, mode);
 
 	rc = dfs_open(parent->ie_dfs->dfs_ns, parent->ie_obj, name,
-		      mode | S_IFDIR, O_CREAT | O_RDWR,
-		      0, 0, NULL, &ie->ie_obj);
-	if (rc)
+		      mode, O_CREAT, 0, 0, NULL, &ie->ie_obj);
```

Comment:
we probably should, I'll respin.  I removed O_DRWR as it doesn't seem possible to do any I/O on a file until it's been opened but it might interact badly with DAOS-6238 as you say.

I hope to have a PR for that issue tomorrow so maybe we could land that first and re-test this once that's landed.

Reply:
Sounds good to me. I do think this _**should**_ work without O_RDWR, so DAOS-6238 would give a more proper fix.